validator_worker/src/follower.rs

```diff
@@ -333,3 +333,527 @@ async fn on_error<'a, C: Unlocked + 'static>(
+            .find_chain_of(DUMMY_CAMPAIGN.channel.token)
+            .expect("Should find Dummy campaign token in config")
+            .with_channel(DUMMY_CAMPAIGN.channel);
+
+        let get_initial_balances = || {
+            let mut balances: Balances<CheckedState> = Balances::new();
+            balances
+                .spend(*ADVERTISER, *PUBLISHER, UnifiedNum::from_u64(1000))
+                .expect("should spend");
+            balances
+                .spend(*ADVERTISER, *PUBLISHER_2, UnifiedNum::from_u64(1000))
+                .expect("should spend");
+            balances
+                .spend(*GUARDIAN, *PUBLISHER, UnifiedNum::from_u64(1000))
+                .expect("should spend");
+            balances
+                .spend(*GUARDIAN, *PUBLISHER_2, UnifiedNum::from_u64(1000))
+                .expect("should spend");
+            balances
+        };
+
+        let get_initial_spenders = || {
+            let mut spenders: HashMap<Address, Spender> = HashMap::new();
+            spenders.insert(
+                *ADVERTISER,
+                Spender {
+                    total_deposited: UnifiedNum::from_u64(10_000),
+                    total_spent: Some(UnifiedNum::from_u64(2000)),
+                },
+            );
+            spenders.insert(
+                *GUARDIAN,
+                Spender {
+                    total_deposited: UnifiedNum::from_u64(10_000),
+                    total_spent: Some(UnifiedNum::from_u64(2000)),
+                },
+            );
+            spenders
+        };
+
+        // Case where all_spenders_sum overflows
+        {
+            let mut all_spenders: HashMap<Address, Spender> = HashMap::new();
+
+            all_spenders.insert(
+                *ADVERTISER,
+                Spender {
+                    total_deposited: UnifiedNum::from_u64(u64::MAX),
+                    total_spent: None,
+                },
+            );
+            all_spenders.insert(
+                *GUARDIAN,
+                Spender {
+                    total_deposited: UnifiedNum::from_u64(u64::MAX),
+                    total_spent: None,
+                },
+            );
+
+            let tick_res = tick(
+                &sentry,
+                &channel_context,
+                all_spenders,
+                get_initial_balances(),
+            )
+            .await;
+            assert!(
+                matches!(tick_res, Err(Error::Overflow)),
+                "Returns an Overflow error"
+            );
+        }
+
+        // - Payout mismatch when checking new_state.balances()
+        {
+            let mut new_state_balances = get_initial_balances().into_unchecked();
+            new_state_balances
+                .earners
+                .insert(*GUARDIAN_2, UnifiedNum::from_u64(10000));
+            let new_state: NewState<UncheckedState> = NewState {
+                state_root: String::new(),
+                signature: IDS[&*LEADER].to_checksum(),
```

Comment:
There isn't a ready-to-use signature in the dummy values. It would get too complicated to generate one for each test case. I think it would be best if we add a few signatures to the dummy values and use them instead